bolt.gemspec

```diff
@@ -60,15 +60,15 @@ Gem::Specification.new do |spec|
   spec.add_dependency "puppetfile-resolver", ">= 0.6.2", "< 1.0"
   spec.add_dependency "puppet-resource_api", ">= 1.8.1"
   spec.add_dependency "puppet-strings", ">= 2.3.0", "< 5.0"
-  spec.add_dependency "r10k", "~> 3.10"
+  spec.add_dependency "r10k", "~> 4.0"
```

Comment:
We cant drop support for 3. We can update to support both 3 and 4 if that is helpful.

Reply:
Hi,
I took a bit of a break in responding here because the whole Ruby situation in the Puppet/Perforce tooling is a bit frustrating.

In theory we could do `spec.add_dependency "r10k", ">= 3.10", "< 5"`. But bundler showed in the past that it had some trouble to figure out correct versions / it didn't always hounor the minimal ruby version in the dependencies. I don't know if this keeps working on ruby 2.5 / 2.6 if we do the change.

Reply:
`spec.add_dependency "r10k", ">= 3.10", "< 5"` seems reasonable to me

Reply:
But in order to actuallly use the new r10k version, we need a release of https://github.com/puppetlabs/orchestrator_client-ruby/pull/38 .

Reply:
Yep.